### PR TITLE
Document `where` support for AR `normalizes`

### PR DESCRIPTION
**Pull request by Earlopain**
### Motivation / Background

Reading through the changelog for `normalized` made me wonder if it also works with `where`.  It says it works  with finder methods, but I don't think `where` is one of those. It works regardless, so I added it to the changelog and made some tests for it.

I also noticed some inconsistencies in the changelog, fixed those up too. I hope that's ok to put in one PR.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Review by akhilgkrishnan — COMMENTED** [review]
@Earlopain Thanks for updating this 👍 . This LGTM after squashing your commits.

**Comment by Earlopain** [comment]
Done!

File: activerecord/CHANGELOG.md [hunk 1]
```diff
@@ -856,6 +856,7 @@
       ```ruby
       class User < ActiveRecord::Base
         normalizes :email, with: -> email { email.strip.downcase }
+        normalizes :phone, with: -> phone { phone.delete("^0-9").delete_prefix("1") }
```
Comment (zzak) [inline activerecord/CHANGELOG.md:859]: @Earlopain This example is a little confusing, how about just `phone.strip`?

Comment (zzak) [reply activerecord/CHANGELOG.md:859]: D'oh, sorry I was mixing up the previous example because you added the `where` below 🤦 sorry!

**Pull request merged**